choreography/src/choreography/Fabrikam.Choreography.ChoreographyService/Controllers/ChoreographyController.cs

```diff
@@ -183,22 +186,20 @@ public async Task<IActionResult> Operation([FromBody] EventGridEvent[] events)
                             }
                         }
                 }
-
-
-
             }
-
-
             return BadRequest();
         }
 
-    }
-}
-
-
-   
-
-     
+        private bool IsDeliveryObjectValid(BinaryData bdata)
+        {
+            JSchemaGenerator generator = new JSchemaGenerator();
+            JSchema schema = generator.Generate(typeof(Delivery));
+            schema.AllowAdditionalPropertiesSpecified = false;
+            schema.AllowAdditionalProperties = false;
```

Comment:
This feels like schema generation should happen outside the mainline processing here.  It's a "do once" type action, not "do for each message." -- Validation, yes, but schema generation, probably not.

Reply:
Good catch. Moved schema generation to its own method and now its called from outside the for loop

Reply:
That's better, but you're still in doing this _per HTTP Post_, eek.  I should have been clearer, sorry, my concern was that this is happening for each and every invocation of this http post, instead of once for the lifetime of the server.  But the fact that it's pulled out of the loop at least makes it a bit better.  We'll just go with it for now.

@magrande